appinventor/buildserver/src/com/google/appinventor/buildserver/tasks/android/RunD8.java

```diff
@@ -66,6 +67,18 @@ public TaskResult execute(AndroidCompilerContext context) {
         inputs.add(preDexLibrary(context, new File(lib)));
       }
 
+      // Add extension libraries
+      Set<String> addedExtJars = new HashSet<>();
+      for (String type : context.getExtCompTypes()) {
+        String sourcePath = ExecutorUtils.getExtCompDirPath(type, context.getProject(),
+            context.getExtTypePathCache())
+            + context.getResources().getSimpleAndroidRuntimeJarPath();
+        if (!addedExtJars.contains(sourcePath)) {
+          inputs.add(new File(sourcePath));
```

Comment:
We should determine whether it makes sense to predex the extension libraries or just let people using extensions pay a performance penalty here.

Reply:
If we predex, will that add their extensions to the dexcache, which if so, would add clutter to the buildserver container?

Reply:
Yes that is correct.  Although I think this was also true previously before we switched to using multiidex (which doesn't support pre dexing).